tests/benchdnn/graph/input_displacer.cpp

```diff
@@ -536,19 +609,49 @@ int partition_data_displacer_t::gen_quantize_filling(
         }
     }
 
-    ::std::unordered_set<size_t> empty_set;
-    const auto &eng = get_test_engine();
-    ref_primitive_t ref_prim(op);
-    ref_prim.init_prb(res);
-    if (res->state == INVALID_ARGUMENTS) return FAIL;
-    SAFE_V(ref_prim.init_prim(eng, res, /* force_override = */ true));
-    if (res->state == SKIPPED || res->state == UNIMPLEMENTED) return OK;
-    ref_prim.init_memory_args(eng);
-    SAFE_V(ref_prim.init_ref_memory_args(eng, res));
-    if (res->state == SKIPPED || res->state == UNIMPLEMENTED) return OK;
+    auto ref_prim = ::std::make_shared<ref_primitive_t>(op);
+    init_ref_prim_and_fill_data(ref_prim, res);
+
+    auto &gen_mem = const_cast<dnn_mem_t &>(ref_prim->get_arg(arg));
+    if (s8_mem_for_u8_wei) {
+        // If s8 data is directly read using the u8 data type, it may lead to
+        // overflow issues. For complex patterns like SDPA, this could result
+        // in precision degradation. Using a reorder to convert negative values
+        // into zeros.
+        dnn_mem_t gen_u8_mem(gen_mem, dnnl_u8, tag::abx, gen_mem.engine());
+        mem = ::std::move(gen_u8_mem);
+    } else
+        mem = ::std::move(gen_mem);
+
+    // Reduce data range to avoid false positive results
+    static constexpr int64_t chunk_size = 64;
```

Comment:
```suggestion
    // Reduce data range to avoid false positive results.
    // Note: traversing over the mem data twice, which is bad  for
    // performance but doesn't require dealing with external
    // data filling configuration.
    static constexpr int64_t chunk_size = 64;
```